### DISCUSSION TITLE
react-spring and IE 11 compatibility guide

### DISCUSSION THREAD

**Discussion by caderitter** [pmndrs/react-spring#1451] [Support]
I thought I'd put this here since we just solved this as a team and thought others might run into the same thing. This may be worth putting somewhere in the updated docs.

React-spring v9 doesn't support IE 11, so you must transpile it by removing react-spring from your babel-loader exclude rule in webpack:

```javascript
{
    test: /\.js$/,
    exclude: /node_modules(?!\/(@react-spring))/,
    loader: 'babel-loader',
},
```

However, we were still finding that ES6 features were appearing in our bundle. We found that `rafz` package was the source of these ES6 features breaking IE 11. It also turns out that the rafz file being used was of type `.mjs`, not `.js`. With a quick change:

```javascript
{
    test: /\.m?js$/,
    exclude: /node_modules(?!\/(@react-spring|rafz))/,
    loader: 'babel-loader',
}
```

It now works in IE 11 🎉

**Comment by joshuaellis** [comment]
Would you be interested in adding this to the docs over on react-spring.io? Also the rafz thing we could probably fix as that's a package in our org.

**Reply by caderitter** [reply]
Certainly!
